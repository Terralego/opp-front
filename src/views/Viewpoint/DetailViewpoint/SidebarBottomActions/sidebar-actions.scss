@import '../../../../variables';

.sibebarActions {
  position: absolute;
  bottom: 1px;
  left: 0;
  right: 0;
  background: $activetabgray;
  border-top: 1px solid #979fa5;
  box-shadow: 0 0 0 1px rgba(16, 22, 26, 0.1), 0 0 0 rgba(16, 22, 26, 0),
    0 -2px 1px rgba(16, 22, 26, 0.2);

  .bp3-navbar {
    &-group {
      display: flex;
      justify-content: space-between;

      .bp3-spinner {
        margin: 0 auto;
      }

      button {
        background: $buttonBlue;
        color: #ffffff;

        .bp3-icon {
          color: #fff;
        }

        &:hover,
        &:focus {
          text-decoration: none;
          color: #ffffff;
        }
      }
    }
  }
}
