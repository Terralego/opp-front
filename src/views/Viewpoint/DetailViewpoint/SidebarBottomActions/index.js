import React from 'react';
import {
  Alignment,
  Button,
  Classes,
  Navbar,
  NavbarGroup,
} from '@blueprintjs/core';
import { useTranslation } from 'react-i18next';

import DownloadButton from '../../../../components/DownloadButton';
import { connectViewpointProvider } from '../../context';

import './sidebar-actions.scss';

export const SidebarActions = ({ viewpoint: { id, label } = {} }) => {
  const { t } = useTranslation();
  return (
    <Navbar className="sibebarActions">
      <NavbarGroup className="navTop__viewpoint-detail" align={Alignment.CENTER}>
        <DownloadButton
          as={Button}
          endpoint={`viewpoints/${id}/pdf`}
          filename={`${label}.pdf`}
          size={25}
          className={Classes.MINIMAL}
          icon="document"
        >
          {t('viewPoint.detail.download.PDF')}
        </DownloadButton>
        <DownloadButton
          as={Button}
          endpoint={`viewpoints/${id}/zip-pictures/`}
          filename={`${label}.zip`}
          size={25}
          className={Classes.MINIMAL}
          icon="cloud-download"
        >
          {t('viewPoint.detail.download.photos')}
        </DownloadButton>
      </NavbarGroup>
    </Navbar>
  );
};

export default connectViewpointProvider('viewpoint')(SidebarActions);
