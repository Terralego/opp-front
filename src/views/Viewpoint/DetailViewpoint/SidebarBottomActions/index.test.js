import React from 'react';
import renderer from 'react-test-renderer';

import SidebarBottomActions from '.';

jest.mock('react-router-dom', () => ({
  Link: () => null,
}));

jest.mock('@blueprintjs/core', () => ({
  Navbar: ({ children }) => children,
  NavbarGroup: ({ children }) => children,
  Alignment: () => null,
  Classes: () => null,
  Button: () => <p>Button</p>,
}));

it('should render', () => {
  const tree = renderer.create(<SidebarBottomActions />).toJSON();
  expect(tree).toMatchSnapshot();
});
