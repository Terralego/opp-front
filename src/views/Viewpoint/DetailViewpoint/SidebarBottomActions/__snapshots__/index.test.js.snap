// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
Array [
  <p>
    Button
  </p>,
  <p>
    Button
  </p>,
]
`;
